File: taskrouter/v1/task/read-example-2/read-example-2.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using System.Collections.Generic;
using Twilio;
using Twilio.Rest.Taskrouter.V1.Workspace;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var assignmentStatus = new List<string> {
            "pending"
        };

        var tasks = TaskResource.Read(
            assignmentStatus: assignmentStatus,
            pathWorkspaceSid: "WSXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
        );

        foreach(var record in tasks)
        {
           Console.WriteLine(record.Sid);
        }
    }
}

File: taskrouter/v1/task/read-example-2/read-example-2.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

tasks = @client.taskrouter
               .workspaces('WSXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
               .tasks
               .list(assignment_status: 'pending')

tasks.each do |record|
  puts record.sid
end
